src/api/layer.js

```diff
@@ -376,12 +381,25 @@ export default class Layer {
     }
 
     _addToMGLMap(map, beforeLayerID) {
-        if (map.isStyleLoaded()) {
-            this._onMapLoaded(map, beforeLayerID);
-        } else {
-            map.on('load', () => {
-                this._onMapLoaded(map, beforeLayerID);
-            });
+        map.once('data', (event) => {
```

Comment:
I don't understand why does this work? Does it work? I thought that MGL required the style to be loaded before the `addLayer` method works

Reply:
MGL requires the style to be loaded if the data event type is sytle. About the `data` event, after reading the documentation I think it is the only change that needs to be listened, and only the first time (that is why I am using `once`). 

https://www.mapbox.com/mapbox-gl-js/api/#map.event:data
> Fired when any map data loads or changes

The examples in the editor work, but I agree it needs further testing.